pennylane/vqe/vqe.py

```diff
@@ -120,6 +120,52 @@ def __str__(self):
 
         return "\n+ ".join(terms)
 
+    def map_wires(self, wire_map):
+        r""" Maps the wires of a Hamiltonian to a new set of wires
+
+        Args:
+           wire_map (dict) : A dictionary that assigns new wires to a subset of the wires tracked in the Hamiltonian
+
+        Returns:
+            ~.Hamiltonian:
+        """
+
+        obs = {
+            "Identity": qml.Identity,
+            "PauliX": qml.PauliX,
+            "PauliY": qml.PauliY,
+            "PauliZ": qml.PauliZ,
+            "Hadamard": qml.Hadamard,
+            "Hermitian": qml.Hermitian,
+        }
+
+        if not isinstance(wire_map, dict):
+            raise ValueError(
+                "wire_map must be of type dict, got {}".format(type(wire_map).__name__)
+            )
+
+        f = lambda wire: wire_map[wire] if wire in wire_map.keys() else wire
+        new_terms = []
+
+        for term in self.ops:
+
+            tensor_term = []
+            term = qml.operation.Tensor(term) if isinstance(term.name, str) else term
+
+            for i in term.obs:
+
+                name = i.name
+                new_wires = list(map(f, i.wires))
```

Comment:
Instead of using `list` and `map`, you could perhaps instead do
```suggestion
                new_wires = [wire_map.get(wire, wire) for wire in i.wires]
```

where `wire_map.get(key, default)` simply returns `default` if the key doesn't exist. This is slightly more 'pythonic' than the functional approach